src/go/cmd/metadata-server/main.go

```diff
@@ -194,10 +195,12 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err := PatchCorefile(ctx, k8s); err != nil {
-		removeIPTablesRule()
-		slog.Error("PatchCorefile", slog.Any("Error", err))
-		os.Exit(1)
+	if !*runningOnGKE {
+		if err := PatchCorefile(ctx, k8s); err != nil {
+			removeIPTablesRule()
+			slog.Error("PatchCorefile", slog.Any("Error", err))
+			os.Exit(1)
+		}
```

Comment:
Do you also want to skip 'RevertCorefile(ctx, k8s)' with this condition?

Reply:
Yes, thanks - a bit of copy-pasting but worth it to avoid generating errors on shutdown in GKE.